Mathlib/Topology/Algebra/Module/Basic.lean

```diff
@@ -0,0 +1,2695 @@
+def Submodule.topologicalClosure (s : Submodule R M) : Submodule R M :=
+  { s.toAddSubmonoid.topologicalClosure with
+    carrier := closure (s : Set M)
+    smul_mem' := s.mapsTo_smul_closure }
+#align submodule.topological_closure Submodule.topologicalClosure
+
+@[simp]
+theorem Submodule.topologicalClosure_coe (s : Submodule R M) :
+    (s.topologicalClosure : Set M) = closure (s : Set M) :=
+  rfl
+#align submodule.topological_closure_coe Submodule.topologicalClosure_coe
+
+theorem Submodule.le_topologicalClosure (s : Submodule R M) : s ≤ s.topologicalClosure :=
+  subset_closure
+#align submodule.le_topological_closure Submodule.le_topologicalClosure
+
+theorem Submodule.isClosed_topologicalClosure (s : Submodule R M) :
+    IsClosed (s.topologicalClosure : Set M) := isClosed_closure
+#align submodule.is_closed_topological_closure Submodule.isClosed_topologicalClosure
+
+theorem Submodule.topologicalClosure_minimal (s : Submodule R M) {t : Submodule R M} (h : s ≤ t)
+    (ht : IsClosed (t : Set M)) : s.topologicalClosure ≤ t :=
+  closure_minimal h ht
+#align submodule.topological_closure_minimal Submodule.topologicalClosure_minimal
+
+theorem Submodule.topologicalClosure_mono {s : Submodule R M} {t : Submodule R M} (h : s ≤ t) :
+    s.topologicalClosure ≤ t.topologicalClosure :=
+  closure_mono h
+#align submodule.topological_closure_mono Submodule.topologicalClosure_mono
+
+/-- The topological closure of a closed submodule `s` is equal to `s`. -/
+theorem IsClosed.submodule_topologicalClosure_eq {s : Submodule R M} (hs : IsClosed (s : Set M)) :
+    s.topologicalClosure = s :=
+  SetLike.ext' hs.closure_eq
+#align is_closed.submodule_topological_closure_eq IsClosed.submodule_topologicalClosure_eq
+
+/-- A subspace is dense iff its topological closure is the entire space. -/
+theorem Submodule.dense_iff_topologicalClosure_eq_top {s : Submodule R M} :
+    Dense (s : Set M) ↔ s.topologicalClosure = ⊤ := by
+  rw [← SetLike.coe_set_eq, dense_iff_closure_eq]
+  simp
+#align submodule.dense_iff_topological_closure_eq_top Submodule.dense_iff_topologicalClosure_eq_top
+
+instance Submodule.topologicalClosure.completeSpace {M' : Type _} [AddCommMonoid M'] [Module R M']
+    [UniformSpace M'] [ContinuousAdd M'] [ContinuousConstSMul R M'] [CompleteSpace M']
+    (U : Submodule R M') : CompleteSpace U.topologicalClosure :=
+  isClosed_closure.completeSpace_coe
+#align submodule.topological_closure.complete_space Submodule.topologicalClosure.completeSpace
+
+/-- A maximal proper subspace of a topological module (i.e a `Submodule` satisfying `IsCoatom`)
+is either closed or dense. -/
+theorem Submodule.isClosed_or_dense_of_isCoatom (s : Submodule R M) (hs : IsCoatom s) :
+    IsClosed (s : Set M) ∨ Dense (s : Set M) := by
+  refine (hs.le_iff.mp s.le_topologicalClosure).symm.imp ?_ dense_iff_topologicalClosure_eq_top.mpr
+  exact fun h ↦ h ▸ isClosed_closure
+#align submodule.is_closed_or_dense_of_is_coatom Submodule.isClosed_or_dense_of_isCoatom
+
```

Comment:
The diff is showing that `isClosed_or_dense_ker` has been deleted here, but this lemma is not in Lean3. Has the Lean3 file changed since the mathport output?

Reply:
Yes, `is_closed_or_dense_ker` was yeeted to `topology/algebra/module/simple` in leanprover-community/mathlib#18608.